api_tests/base/test_sloan_study.py

```diff
@@ -141,6 +152,7 @@ def test_sloan_study_control_unauth(self, app, user, preprint):
 
     @pytest.mark.parametrize('reffer_url, expected_provider_id', [
         (f'https://staging2.osf.io/', None),
+        (f'https://staging2.osf.io/ispp0/', 'test id'),
```

Comment:
In some cases we do, in some cases not. `https://staging2-engrxiv.cos.io/hnwfp` for example does not redirect `https://staging2.osf.io/preprints/metaarxiv/t8wf7/` does. But `staging2-engrxiv` should resolve to the `engrxiv` provider.

Reply:
But that's because `staging2-engrxiv.cos.io` is a custom domain, which should be covered by your burdrxiv tests. I don't think it hurts to have this case covered, but I'm not sure under what circumstances it gets hit.

Reply:
Different staging environments are just set up differently in this regards. For example `https://api.staging.osf.io/v2/providers/preprints/osf/` the staging osf providers domain is `http://osf.io`, but https://api.staging2.osf.io/v2/providers/preprints/osf/ domain is `https://staging2.osf.io/` ie the domain of the staging provider, not the `http://osf.io`

Reply:
Innnteresting.